.github/workflows/pending-clear.yml

```diff
@@ -0,0 +1,30 @@
+name: "clear pending status"
+
+on:
+  check_suite:
+    types: [ completed ]
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    steps:
+    - name: clear pending status
+      if: github.repository_owner == 'NixOS' && github.event.check_suite.app.name == 'OfBorg'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GSU_VERSION: "0.5.0"
+        GSU_URL: "https://github.com/cloudposse/github-status-updater/releases/download"
```

Comment:
A bit late, but I think this should be packaged in Nixpkgs and used from there, so there's less of a chance of something happening to the release (either being removed somehow, or being re-uploaded with the ability to exfiltrate the `$GITHUB_TOKEN`, etc.). Seems wrong to me that we fetch from a URL we have no control over, when we use hashes basically everywhere in Nixpkgs to mostly guard against bad actors ("TOFU" notwithstanding).

The `$GITHUB_TOKEN` token has some potentially-abusable permissions in the worst case (https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#permissions-for-the-github_token), such as r/w to issues, PRs, and projects.

(I'm no expert on GitHub nor its APIs, so maybe this isn't a problem?)

Reply:
> A bit late, but I think this should be packaged in Nixpkgs and used from there, so there's less of a chance of something happening to the release (either being removed somehow, or being re-uploaded with the ability to exfiltrate the $GITHUB_TOKEN, etc.). Seems wrong to me that we fetch from a URL we have no control over, when we use hashes basically everywhere in Nixpkgs to mostly guard against bad actors ("TOFU" notwithstanding).

This is more expensive as it requires installing nix.

We could try caching them or mirroring them somewhere.

> The $GITHUB_TOKEN token has some potentially-abusable permissions

It's using the default token.

Reply:
> The $GITHUB_TOKEN token has some potentially-abusable permissions

Sorry, meant to say it's using `pull_request_target` so the workflow would need to be modified on the base branch.

Reply:
We could just compute the checksum before unpacking the tarball.

Reply:
Added a hash check: https://github.com/NixOS/nixpkgs/commit/c5dc3692b8ce0037c7004f9e7aec3af206056244

We may want to use https://github.com/actions/github-script or just switch to a raw `curl` command.

Reply:
Yeah github-script looks nice too.

Reply:
Sorry, forgot to mention here that I changed it to curl.

64a2de1fc0c16b1a5ff9afe8c85aeb4dcfa019bb

Reply:
I'm much happier with that, thanks :)